zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationConfig.java

```diff
@@ -141,8 +143,10 @@ public static X509AuthenticationConfig getInstance() {
   //creation of multiple objects; not using here for the consideration of read performance
   private Set<String> openReadAccessPathPrefixes;
   private Set<String> crossDomainAccessDomains;
+  private Set<String> znodeGroupAclSuperUserIds;
   private final Object openReadAccessPathPrefixesLock = new Object();
   private final Object crossDomainAccessDomainsLock = new Object();
+  private final Object znodeGroupAclSuperUserIdsLock = new Object();
```

Comment:
Just curious why all these setters are protected by lock?

Reply:
As my above reply, we are using double checked locking mechanism to do lazy instantiation of these fields.